mockgen/mockgen.go

```diff
@@ -130,7 +130,11 @@ func main() {
 	// is output into an already existing package.
 	outputPackagePath := *selfPackage
 	if outputPackagePath == "" && *destination != "" {
-		dstPath, err := filepath.Abs(filepath.Dir(*destination))
+		currentDir,err:=os.Getwd()
+		if err != nil {
+			log.Fatalf("%v",err)
+		}
+		dstPath, err := filepath.Abs(filepath.Dir(strings.Join([]string{currentDir,*destination},string(os.PathSeparator))))
```

Comment:
Isn't this what `filepath.Abs()` [already does](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/path/filepath/path.go;l=302) if its input isn't an absolute path? Can you give some examples of what you're trying to achieve, please?

Separately, `filepath.Join()` should be used over `strings.Join()` in this case.